### PR TITLE
Allow to prepare additional caches

### PR DESCRIPTION
**Pull request by snicoll**
Update `CouchbaseCacheManager` to offer additional customizations on
startup and in particular the ability to prepare (register) additional
caches on startup.

### CODE REVIEW

File: src/main/java/com/couchbase/client/spring/cache/CouchbaseCacheManager.java [hunk 1]
```diff
@@ -101,6 +103,39 @@ public void setDefaultCacheBuilder(CacheBuilder defaultCacheBuilder) {
     this.defaultCacheBuilder = defaultCacheBuilder;
   }
 
+  /**
+   * Register an additional cache with the specified name using the default builder.
+   * <p>
+   * Caches can only be configured at initialization time. Once the cache manager
+   * has been initialized, no cache can be further prepared.
+   * @param name the name of the cache to add
+   * @throws IllegalStateException if no builder is available or if the cache manager
```
Comment (simonbasle) [inline src/main/java/com/couchbase/client/spring/cache/CouchbaseCacheManager.java]: would maybe phrase it like "if the CacheManager isn't dynamic (no builder available)"...

File: src/main/java/com/couchbase/client/spring/cache/CouchbaseCacheManager.java [hunk 2]
```diff
@@ -112,6 +147,7 @@ protected Cache getMissingCache(String name) {
     for (Map.Entry<String, CacheBuilder> entry : initialCaches.entrySet()) {
       caches.add(entry.getValue().build(entry.getKey()));
     }
+    initialized = true;
```
Comment (simonbasle) [inline src/main/java/com/couchbase/client/spring/cache/CouchbaseCacheManager.java]: the initialized flag should be set to true above the usage of `initialCaches`

Comment (snicoll) [inline src/main/java/com/couchbase/client/spring/cache/CouchbaseCacheManager.java]: Probably, yeah.

File: src/main/java/com/couchbase/client/spring/cache/CouchbaseCacheManager.java (hunk 1 above)
Comment (snicoll) [inline src/main/java/com/couchbase/client/spring/cache/CouchbaseCacheManager.java]: I don't know if you use that dynamic term elsewhere but feel free to update the doc any way you like.

**Comment by simonbasle** [comment]
:+1:  looks good to me

**Pull request merged**